internal/mic/mic.go

```diff
@@ -50,6 +51,8 @@ func (mici *micImpl) CreateOrPatch(ctx context.Context, name, ns string, images
 		},
 	}
 
+	images = removeDuplicateImages(images)
```

Comment:
Can we use a Set directly when [creating the imageList](https://github.com/kubernetes-sigs/kernel-module-management/blob/189d136a41e17852ab54f197a9487ee46a9fc591/internal/controllers/module_reconciler.go#L376) instead of filtering it later on?